tests/testthat/test-incidence.R

```diff
@@ -0,0 +1,154 @@
+test_that("graph_from_biadjacency_matrix() works -- dense", {
+  local_igraph_options(print.id = FALSE)
+  withr::local_seed(42)
+
+  inc <- matrix(sample(0:1, 15, repl = TRUE), 3, 5)
+  colnames(inc) <- letters[1:5]
+  rownames(inc) <- LETTERS[1:3]
+
+  expect_snapshot((g <- graph_from_biadjacency_matrix(inc)))
+  expect_false(is_weighted(g))
+
+  expect_snapshot((weighted_g <- graph_from_biadjacency_matrix(inc, weighted = TRUE)))
+  expect_true(is_weighted(weighted_g))
+})
+
+test_that("graph_from_biadjacency_matrix() works - dense, modes", {
+  local_igraph_options(print.id = FALSE)
+  withr::local_seed(42)
+
+  inc <- matrix(sample(0:1, 15, repl = TRUE), 3, 5)
+  colnames(inc) <- letters[1:5]
+  rownames(inc) <- LETTERS[1:3]
+
+  out_g <- graph_from_biadjacency_matrix(inc, directed = TRUE, mode = "out")
+  expect_true(is_directed(out_g))
+  expect_length(E(out_g), 7)
+  expect_equal(as_adj_list(out_g, mode = "out")$A %>% as.numeric(), c(6, 7))
+
+  in_g <- graph_from_biadjacency_matrix(inc, directed = TRUE, mode = "in")
+  expect_true(is_directed(in_g))
+  expect_length(E(in_g), 7)
+  expect_equal(as_adj_list(in_g, mode = "in")$A %>% as.numeric(), c(6, 7))
+
+  total_g <- graph_from_biadjacency_matrix(inc, directed = TRUE, mode = "total")
+  expect_true(is_directed(total_g))
+  expect_length(E(total_g), 14)
+  expect_equal(as_adj_list(total_g, mode = "all")$A %>% as.numeric(), c(6, 6, 7, 7))
```

Comment:
Can you please use `mode = "all"` here, and rename `total_g` to `mutual_g`?

With `mode`, the strings `"all"` and `"total"` are always treated as aliases. But the terminology of "total" only makes sense when talking about degrees. In a context where we talk about edge directions, this word would never be used.  In igraph, we generally call `<->` edges "mutual".